lib/providers/fs.js

```diff
@@ -71,7 +78,14 @@ module.exports = function(options) {
     var result = {};
     var foundPath;
     var fullPathnames = publicPaths.map(function(p) {
-      return pathjoin(cwd, p, pathname);
+      var bp = pathjoin(cwd, p);
+      var fp = pathjoin(cwd, p, pathname);
+      if (path.relative(bp, fp).indexOf('..') === 0) {
+        return null;
+      }
+      return fp;
+    }).filter(function(p) {
+      return !!p;
```

Comment:
This whole function is quite terse - the short variables names make it slow to read through and verify correctness. Also,  `!!` is one of my least favorite expressions in JavaScript - isn't this basically a null check? Why not use `_.startsWith(relativePath, '..')` and `!_.isNull(p)` to be a little more descriptive - the filesystem lookups are always going to be the slowest part of this method